### ISSUE TITLE
ClassNotFoundException for DropwizardTestResourceConfig

### ISSUE THREAD

**Issue by haasted** [dropwizard/dropwizard#2398]
I'm encountering a strange error during a very simple scenario:

Attempting to use a `ResourceTestRule` to verify the behaviour of a resource class. 

The test is set up as follows, which is an almost exact replica of the example given in the documentation:

```
public class HelloWorldTest {
    @ClassRule
    public static final ResourceTestRule resources = ResourceTestRule.builder()
            .addResource(new HelloWorld())
            .build();

    @Test
    public void test1() throws IOException {
        String s = resources.target("/hello-world").request().get(String.class);
        System.out.println(s);
    }
}
```
This results in the following error during test setup: 

```
java.lang.NullPointerException
  at java.util.Objects.requireNonNull(Objects.java:203)
  at io.dropwizard.testing.common.Resource.after(Resource.java:213)
  at io.dropwizard.testing.junit.ResourceTestRule$1.evaluate(ResourceTestRule.java:102)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
  at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
  at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
  at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
  at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
  at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```
With the aid of an exception breakpoint, I tracked down the actual error, which appears to a failure to load the class `io.dropwizard.testing.common.DropwizardTestResourceConfig`, somewhere in the `io.dropwizard.testing.common.Resource#before` method.

I'm at my wits end. Can anyone replicate this issue, or have a suggestion as to how it can be resolved?

My DW dependencies are as follows:

    compile "io.dropwizard:dropwizard-core:1.3.2"
    compile "io.dropwizard:dropwizard-jdbi3:1.3.2"
    compile "io.dropwizard:dropwizard-migrations:1.3.2"
    compile "io.dropwizard:dropwizard-testing:1.3.2"

I do realize that the final dependency should probably be `testCompile`. I've attempted an upgrade to 1.3.4, but this did not impact the issue.

**Comment by joschi** [comment]
@haasted Could you push a minimal example to reproduce the issue?

**Comment by haasted** [comment]
I'm embarrassed to say that it's somehow working now, seemingly without any modifications. Must have had a bad day.

Apologies for the inconvenience, and thanks for putting together a great stack!

**Comment by haasted** [comment]
Minor update as I found the source of the issue.

I had experimented with using Grizzly as a test container, but unfortunately neglected to depend on the same version imported by DW. 

It seems that merely the presence of the incorrect `jersey-test-framework-provider-grizzly' on the classpath was enough to trigger the issue, i.e. adding the following caused the problem:

    testCompile 'org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-grizzly2:2.27'

Changing the version to 2.25.1 fixed the issue.
